newIDE/app/src/EventsFunctionsExtensionEditor/EventsFunctionConfigurationEditor/EventsFunctionParametersEditor.js

```diff
@@ -72,6 +74,80 @@ const validateParameterName = (i18n: I18nType, newName: string) => {
   return true;
 };
 
+type StrngSelectorEditorProps = {|
+  extraInfo: string,
+  setExtraInfo: newExtraInfo => void,
+|};
+
+const StringSelectorEditor = ({
+  extraInfo,
+  setExtraInfo,
+}: StrngSelectorEditorProps) => {
+  const [array, setArray] = React.useState([]);
+  const forceUpdate = useForceUpdate();
+  React.useEffect(
+    () => {
+      let parsedArray = null;
+      try {
+        parsedArray = JSON.parse(extraInfo);
+      } catch (e) {
+        console.warn('Cannot parse parameter extraInfo: ', e);
+      }
+
+      if (parsedArray !== null) {
+        setArray(parsedArray);
+      }
+    },
+    [extraInfo, array]
+  );
```

Comment:
I'm unsure about this effect. It depends on `array`, but you call `setArray` inside, so every time it's run, it will be run again because `setArray` will be called, so array will change?

I think we don't need this complexity and don't need a state, because we're deriving some data from some other data, which we don't need to store:

```suggestion
  let array = [];
  try {
    array = JSON.parse(extraInfo);
  } catch (e) {
    console.error('Cannot parse parameter extraInfo: ', e);
  }
```